### ISSUE TITLE
Cannot create zip

### ISSUE THREAD

**Issue by drsku** [skoobe/riofs#103]
We have an application that houses thousands of images. Our application is built in java.

We have a process in our application that builds a zip file by adding images, selected by a user, to a temporary folder, then zipping this folder for download.

This fails almost all the time in non-debug mode. Either the images are 0kB in the zip folder or the zip folder ends up being 0kb or does not build. But it works in debug or if we put in lots of delays in the process.

Any suggestions?

Thanks

**Comment by wizzard** [comment]
Hello,

RioFS does not support appending data to an existing file.

**Comment by drsku** [comment]
Does this mean it cannot zip files in a folder?

**Comment by drsku** [comment]
It would be a major limitation if you can't zip files in a folder.

**Comment by drsku** [comment]
Though it is failing on adding files to the temp folder, even before zipping them up.  This doesn't happen in debug mode. You advice would be much appreciate.

**Comment by wizzard** [comment]
In order to further assist you I need to see RioFS debug log file. 
To start RioFS in “verbose” mode please start riofs with “--verbose” command line option and redirect output to a file:  ```riofs --verbose > /tmp/riofs.log``` 
Then reproduce the issue and send me ```/tmp/riofs.log``` file by email. 
Thank you!

**Comment by drsku** [comment]
Hi Paul,
We have debugged the scenario more &  found whenever we upload the file on s3 using riofs it uploads it asynchronously..

We are using java file system to copy/upload the file on s3.

Here is the code snippet:-
org.apache.commons.io. FileUtils.copyInputStreamToFile(inputStream,TargetFile);


In our case we were trying to upload the file of size around 4MB at path /u01/s3fs/media/sys_master/h94/h9b/9073934303262.jpg

After calling the above Java API, immediately control came to the next line without uploading the data on s3.

In order to verify we have also tested with the Unix cp command to upload the data on s3 & found it immediately comes to the next line. Since it was 4MB file so it should take at least some couple of seconds to upload the file & then only control should come to the next line.

So I just want to know if we use riofs then will it upload the data asynchrously & if yes, how can we make it to behave in a synchronous manner..!!

We have emailed you the logs related to the upload operation was written.. 

Does anyone else on this forum have similar issues?

**Comment by StefanCG** [comment]
Yeah we were having issues with files that are being uploaded by end-users and stored on a riofs mounted s3 bucket.

If this is followed by a read operation immediately (like PHP's getimagesize()) this returned a read error. We worked around this by adding a sleep(1) but that's more or less just a dirty fix.

As we suspect that there might be additional issues with this behaviour (when working with certs generated by openssl) we'd also like if there's a way to make riofs uploa dfiles synchronously.
